### ISSUE TITLE
Struct with *time.Time field doesn't import time

### ISSUE THREAD

**Issue by ttacon** [philhofer/msgp#45]
If I have a struct such as:

```go
type A struct {
    C *time.Time
}
```

msgp does not add a time import into the generated code. This raises a compilation error as in Unmarshal/Decode such a field has the following code generated for it:

```go
if identifier == nil {
    idenifier = new(qualified.Type)
}
```

I don't currently have the time to provide a patch for this, but I will when I can find some.

**Comment by philhofer** [comment]
Good catch.

**Comment by ttacon** [comment]
So I mean, I see a couple ways to deal with it.

1) (Sad but very effective) Reprocess the generate file and pass through imports.Process (golang.org/x/tools/imports). The benefit here is it will catch any embedded pointer field of any type (as long as the type exists on the user's GOPATH). Sadly, it's kinda gross - but it could be as simple as a ~10 line change this way and the speed alteration isn't that bad (I literally reread the generated file, passed it through and wrote back to the file).

2) Could keep track of fields identified as pointers and try to identify their containing package. The obvious upside is there's no enormous amount of extra IO but the code complexity is raised a decent amount (especially if a decent attempt is made to resolve the package).

Which way would you be leaning?

**Comment by philhofer** [comment]
I'm fine with (1) for now. We can revisit this if generator time starts getting out of hand. (On my computer, the test file takes 22ms, which is just fine.) FWIW, that's precisely how `gofmt`-ing of the generated source works.

All in all, I think generating one file requires generating no fewer than 4 AST trees. (Ew.)

**Comment by ttacon** [comment]
Just pushed one way of doing it in PR #46

**Comment by philhofer** [comment]
Formally fixed by #46.

**Comment by philhofer** [comment]
I've added #47 in case we want to re-visit this.
